src/agentscope/agents/agent.py

```diff
@@ -13,6 +15,51 @@
 from agentscope.agents.operator import Operator
 from agentscope.models import load_model_by_config_name
 from agentscope.memory import TemporaryMemory
+from agentscope.utils.logging_utils import log_studio
+from agentscope.web.studio.utils import thread_local_data
+from agentscope.web.studio.constants import _SPEAK
+
+
+def pre_reply_decorator(func: Callable) -> Callable:
+    """
+    Decorator for providing additional functionality before the execution
+    of `reply` method.
+
+    This decorator function adds a sleep delay and a "speak" action before
+    calling the actual method it decorates. It checks if a 'uid' attribute
+    is set on the thread's local data.
+
+    Args:
+        func (Callable): The original method that the decorator will wrap.
+
+    Returns:
+        Callable: The decorated/wrapped method.
+    """
+
+    def wrapper(self: Any, *args: Any, **kwargs: Any) -> Any:
+        """
+        The wrapper function that executes the augmented functionality.
+
+        Args:
+            self (Any): The instance of the object to which the method being
+            decorated belongs.
+            *args (Any): Variable length argument list for the decorated
+            method.
+            **kwargs (Any): Arbitrary keyword arguments for the decorated
+            method.
+
+        Returns:
+            Any: The return value from the decorated method.
+        """
+        if hasattr(thread_local_data, "uid"):
+            time.sleep(0.1)
```

Comment:
remove sleep.
Using sleep here will cause a time delay in model calling, further increasing the running time.